src/main/kotlin/no/nav/familie/ef/iverksett/økonomi/simulering/SimuleringUtil.kt

```diff
@@ -29,69 +40,104 @@ fun lagSimuleringsoppsummering(detaljertSimuleringResultat: DetaljertSimuleringR
+                        resultat = hentResultat(posteringListe),
+                        feilutbetaling = hentFeilutbetaling(posteringListe)
+                ).medEtterbetaling(hentEtterbetaling(posteringListe))
+            }
 }
 
 private fun hentNyttBeløp(posteringer: List<SimulertPostering>): BigDecimal {
-    val sumPositiveYtelser = posteringer.filter { postering ->
-        postering.posteringType == PosteringType.YTELSE && postering.beløp > BigDecimal.ZERO
-    }.sumOf { it.beløp }
-    val feilutbetaling = hentFeilutbetaling(posteringer)
-    return if (feilutbetaling > BigDecimal.ZERO) sumPositiveYtelser - feilutbetaling else sumPositiveYtelser
+    val sumPositiveYtelser = posteringer
+            .filter { it.posteringType == YTELSE && it.beløp > ZERO }
+            .sumOf { it.beløp }
+
+    val positivFeilutbetaling = maxOf(hentFeilutbetaling(posteringer), ZERO)
+
+    return sumPositiveYtelser - positivFeilutbetaling
 }
 
 private fun hentFeilutbetaling(posteringer: List<SimulertPostering>) =
-        posteringer.filter { postering ->
-            postering.posteringType == PosteringType.FEILUTBETALING
-        }.sumOf { it.beløp }
+        posteringer
+                .filter { it.posteringType == FEILUTBETALING }
+                .sumOf { it.beløp }
 
 private fun hentTidligereUtbetalt(posteringer: List<SimulertPostering>): BigDecimal {
-    val sumNegativeYtelser = posteringer.filter { postering ->
-        (postering.posteringType === PosteringType.YTELSE && postering.beløp < BigDecimal.ZERO)
-    }.sumOf { -it.beløp }
-    val feilutbetaling = hentFeilutbetaling(posteringer)
-    return if (feilutbetaling < BigDecimal.ZERO) sumNegativeYtelser - feilutbetaling else sumNegativeYtelser
+    val sumNegativeYtelser = posteringer
+            .filter { it.posteringType === YTELSE && it.beløp < ZERO }
+            .sumOf { it.beløp }
+
+    val negativFeilutbetaling = minOf(hentFeilutbetaling(posteringer), ZERO)
+    return negativFeilutbetaling - sumNegativeYtelser
+}
+
+private fun hentResultat(posteringer: List<SimulertPostering>): BigDecimal {
+    val posteringerHarFeilutbetaling = posteringer.any { it.posteringType == FEILUTBETALING }
+    return when {
+        posteringerHarFeilutbetaling -> posteringer
+                .filter { it.posteringType == FEILUTBETALING }
+                .sumOf { -it.beløp }
+        else -> posteringer.sumOf { it.beløp }
+    }
 }
 
-private fun hentResultat(posteringer: List<SimulertPostering>) =
-        if (posteringer.map { it.posteringType }.contains(PosteringType.FEILUTBETALING)) {
-            posteringer.filter {
-                it.posteringType == PosteringType.FEILUTBETALING
-            }.sumOf { -it.beløp }
-        } else
-            posteringer.sumOf { it.beløp }
+private fun hentEtterbetaling(posteringer: List<SimulertPostering>): BigDecimal {
+    val posteringerHarPositivFeilutbetaling =
+            posteringer.any { it.posteringType == FEILUTBETALING && it.beløp > ZERO }
```

Comment:
Kan en feilutbetaling både være positiv og negativ? I hvilke situasjoner skjer det? 

På noen steder i koden håndteres negative tall, og noen ikke, og vet ikke helt om jeg skjønner in/output her (egentlige ikke nytt i denne PR, men sån generellt), burde noen annen som har jobbet med simulering også ta en titt på dette? :)

Reply:
Sliter litt med det samme :D Og ikke minst - hva skjer når noen HAR betalt fakturaen - fjernes disse posteringene eller kommer det inn en ny med feilutbetaling og positivt beløp? 😱

Reply:
Ja, jeg sliter med det samme. Har ikke HELT skjønt hva de ulike posteringene kan være. Prøver å spørre Øystein, jeg